src/main/java/com/yelp/nrtsearch/server/luceneserver/doc/LoadedDocValues.java

```diff
@@ -190,6 +366,10 @@ public SingleDateTime(NumericDocValues docValues) {
       super(docValues, DATE_DECODER);
     }
 
+    public Instant getValue() {
```

Comment:
SingleDataTime is the only implementation of SingleNumericValue now. Do you want to remove the SingleNumericValue class and put its functionality here?

Reply:
I think it is ok to leave `SingleNumericValue` for now, since it would make it easier to add new implementations. Though I am considering breaking these into individual class files, so maybe it can be done then.